### PR TITLE
fix(inline_diff): support ignoring EOL whitespace in inline diff (default: True)

### LINKED ISSUE #177: Inline diff very slow at handling files with different whitespace style
I occasionally encounter this issue when checking out repositories with windows style whitespace; this seems because a whitespace diff is very very slow for file with different style whitespace. This happens only with "inline diff" (`l`) but not with normal diff (`e`). In addition to being slow, the diff shows the entire content as changed (whole file colored red followed by whole file colored green) so it's also basically useless.

To reproduce this, any file that uses `\r\n` as the EOL characters can be used. Open it in a unix style EOL machine and introduce a change. Here is a log for an example with a simple Django `manage.py` file:

### [log from a "simple" diff](https://gist.github.com/cbbdc3d22fa232ed8d6d)

### PR DESCRIPTION
**Pull request by asfaltboy**
close #177

### CODE REVIEW

**Comment by asfaltboy** [comment]
While this code allows properly running an inline diff, trying to stage a line/hunk that removes a line with a line-break (EOL chars) fails with this error:

> `/usr/local/bin/git apply --unidiff-zero --cached -` failed with following output:
> 
> error: patch failed: partnersint/manage.py:7
> error: partnersint/manage.py: patch does not apply

Other lines seem to be staged properly

### log output

    {
      "stdout": "## development\n M proj/manage.py\n", 
      "stderr": "", 
      "command": [
        "status", 
        "-b", 
        "--porcelain"
      ], 
      "type": "git", 
      "stdin": null
    }, 
    {
      "stdout": "96b8510 some commit", 
      "stderr": "", 
      "command": [
        "log", 
        "-n 1", 
        "--pretty=format:%h %s", 
        "--abbrev-commit"
      ], 
      "type": "git", 
      "stdin": null
    }, 
    {
      "stdout": "", 
      "stderr": "", 
      "command": [
        "stash", 
        "list"
      ], 
      "type": "git", 
      "stdin": null
    }, 
    {
      "stdout": "100644 c3845ab520d51f36ccc05a6d02e8160eab563e38 0\tproj/manage.py\n", 
      "stderr": "", 
      "command": [
        "ls-files", 
        "-s", 
        ".../proj/manage.py"
      ], 
      "type": "git", 
      "stdin": null
    }, 
    {
      "stdout": "#!/usr/bin/env python\r\nimport os\r\nimport sys\r\n\r\nif __name__ == \"__main__\":\r\n    os.environ.setdefault(\"DJANGO_SETTINGS_MODULE\", \"settings\")\r\n\r\n    from django.core.management import execute_from_command_line\r\n\r\n    execute_from_command_line(sys.argv)\r\n", 
      "stderr": "", 
      "command": [
        "show", 
        "c3845ab520d51f36ccc05a6d02e8160eab563e38"
      ], 
      "type": "git", 
      "stdin": null
    }, 
    {
      "stdout": "215e14ad2043a5e588834ba539b5bdbb9c9988b9\n", 
      "stderr": "", 
      "command": [
        "hash-object", 
        "-w", 
        "--stdin"
      ], 
      "type": "git", 
      "stdin": "#!/usr/bin/env python\nimport os\nimport sys\n\nif __name__ == \"__main__\":\ntest\n\n    from django.core.management import execute_from_command_line\n\n    execute_from_command_line(sys.argv)\n "
    }, 
    {
      "stdout": "diff --git a/c3845ab520d51f36ccc05a6d02e8160eab563e38 b/215e14ad2043a5e588834ba539b5bdbb9c9988b9\nindex c3845ab..215e14a 100644\n--- a/c3845ab520d51f36ccc05a6d02e8160eab563e38\n+++ b/215e14ad2043a5e588834ba539b5bdbb9c9988b9\n@@ -6 +6 @@ if __name__ == \"__main__\":\n-    os.environ.setdefault(\"DJANGO_SETTINGS_MODULE\", \"settings\")\r\n+test\n@@ -10,0 +11 @@ if __name__ == \"__main__\":\n+ \n\\ No newline at end of file\n", 
      "stderr": "", 
      "command": [
        "diff", 
        "-U0", 
        "--ignore-space-at-eol", 
        "c3845ab520d51f36ccc05a6d02e8160eab563e38", 
        "215e14ad2043a5e588834ba539b5bdbb9c9988b9"
      ], 
      "type": "git", 
      "stdin": null
    }, 
    {
      "stdout": "## development\n M proj/manage.py\n", 
      "stderr": "", 
      "command": [
        "status", 
        "-b", 
        "--porcelain"
      ], 
      "type": "git", 
      "stdin": null
    }, 
    {
      "stdout": "## development\n M proj/manage.py\n", 
      "stderr": "", 
      "command": [
        "status", 
        "-b", 
        "--porcelain"
      ], 
      "type": "git", 
      "stdin": null
    }, 
    {
      "stdout": "## development\n M proj/manage.py\n", 
      "stderr": "", 
      "command": [
        "status", 
        "-b", 
        "--porcelain"
      ], 
      "type": "git", 
      "stdin": null
    }, 
    {
      "stdout": " M proj/manage.py\u0000", 
      "stderr": "", 
      "command": [
        "status", 
        "--porcelain", 
        "-z"
      ], 
      "type": "git", 
      "stdin": null
    }, 
    {
      "stdout": "", 
      "stderr": "error: patch failed: proj/manage.py:7\nerror: proj/manage.py: patch does not apply\n", 
      "command": [
        "apply", 
        "--unidiff-zero", 
        null, 
        "--cached", 
        "-"
      ], 
      "type": "git", 
      "stdin": "diff --git a/proj/manage.py b/proj/manage.py\n--- a/proj/manage.py\n+++ b/proj/manage.py\n@@ -7,1 +8,0 @@\n-    os.environ.setdefault(\"DJANGO_SETTINGS_MODULE\", \"settings\")\n"
    }, 
    {
      "error": "'`/usr/local/bin/git apply --unidiff-zero --cached -` failed with following output:\\n\\nerror: patch failed: proj/manage.py:7\\nerror: proj/manage.py: patch does not apply\\n'", 
      "type": "error"
    }, 
    {
      "stdout": "100644 c3845ab520d51f36ccc05a6d02e8160eab563e38 0\tproj/manage.py\n", 
      "stderr": "", 
      "command": [
        "ls-files", 
        "-s", 
        ".../proj/manage.py"
      ], 
      "type": "git", 
      "stdin": null
    }, 
    {
      "stdout": "#!/usr/bin/env python\r\nimport os\r\nimport sys\r\n\r\nif __name__ == \"__main__\":\r\n    os.environ.setdefault(\"DJANGO_SETTINGS_MODULE\", \"settings\")\r\n\r\n    from django.core.management import execute_from_command_line\r\n\r\n    execute_from_command_line(sys.argv)\r\n", 
      "stderr": "", 
      "command": [
        "show", 
        "c3845ab520d51f36ccc05a6d02e8160eab563e38"
      ], 
      "type": "git", 
      "stdin": null
    }, 
    {
      "stdout": "215e14ad2043a5e588834ba539b5bdbb9c9988b9\n", 
      "stderr": "", 
      "command": [
        "hash-object", 
        "-w", 
        "--stdin"
      ], 
      "type": "git", 
      "stdin": "#!/usr/bin/env python\nimport os\nimport sys\n\nif __name__ == \"__main__\":\ntest\n\n    from django.core.management import execute_from_command_line\n\n    execute_from_command_line(sys.argv)\n "
    }, 
    {
      "stdout": "diff --git a/c3845ab520d51f36ccc05a6d02e8160eab563e38 b/215e14ad2043a5e588834ba539b5bdbb9c9988b9\nindex c3845ab..215e14a 100644\n--- a/c3845ab520d51f36ccc05a6d02e8160eab563e38\n+++ b/215e14ad2043a5e588834ba539b5bdbb9c9988b9\n@@ -6 +6 @@ if __name__ == \"__main__\":\n-    os.environ.setdefault(\"DJANGO_SETTINGS_MODULE\", \"settings\")\r\n+test\n@@ -10,0 +11 @@ if __name__ == \"__main__\":\n+ \n\\ No newline at end of file\n", 
      "stderr": "", 
      "command": [
        "diff", 
        "-U0", 
        "--ignore-space-at-eol", 
        "c3845ab520d51f36ccc05a6d02e8160eab563e38", 
        "215e14ad2043a5e588834ba539b5bdbb9c9988b9"
      ], 
      "type": "git", 
      "stdin": null
    }

File: GitSavvy.sublime-settings [hunk 1]
```diff
@@ -54,6 +54,13 @@
     },
 
     /*
+        Change this to `false` when you would like inline diff to
+        take EOL whitespace changes into account.
+        See details in https://github.com/divmain/GitSavvy/issues/177
+     */
+    "inline_diff_ignore_eol_whitespaces": true,
```
Comment (divmain) [inline GitSavvy.sublime-settings]: So that this is a non-breaking change, default value should be `false`.

File: core/commands/inline_diff.py [hunk 2]
```diff
@@ -122,6 +122,9 @@ class GsInlineDiffRefreshCommand(TextCommand, GitCommand):
     def run(self, edit):
         file_path = self.file_path
         in_cached_mode = self.view.settings().get("git_savvy.inline_diff.cached")
+        ignore_eol_arg = sublime.load_settings("GitSavvy.sublime-settings").get(
+            "inline_diff_ignore_eol_whitespaces", True) and (
+                "--ignore-space-at-eol") or None
```
Comment (divmain) [inline core/commands/inline_diff.py]: Please use the following here:
```python
savvy_settings = sublime.load_settings("GitSavvy.sublime-settings")
ignore_eol_arg = (
    "--ignore-space-at-eol"
    if savvy_settings.get("inline_diff_ignore_eol_whitespaces", True)
    else None
    )
```

File: core/commands/inline_diff.py [hunk 3]
```diff
@@ -375,6 +378,9 @@ def run(self, edit, **kwargs):
 
     def run_async(self, reset=False):
         in_cached_mode = self.view.settings().get("git_savvy.inline_diff.cached")
+        ignore_ws = sublime.load_settings("GitSavvy.sublime-settings").get(
+            "inline_diff_ignore_eol_whitespaces", True) and (
+                "--ignore-whitespace") or None
```
Comment (divmain) [inline core/commands/inline_diff.py]: Same formatting comment as above.

**Comment by divmain** [comment]
This seems like a reasonable way to address the issue you've encountered.  After you make the requested changes, I'll merge this in.  Thanks!

**Comment by asfaltboy** [comment]
Updated and squashed into a single clean commit

**Comment by divmain** [comment]
Looks good - thanks!

**Pull request merged**